pallets/gear/src/tests.rs

```diff
@@ -9652,6 +9652,90 @@ fn test_reinstrumentation_works() {
     })
 }
 
+#[test]
+fn test_reinstrumentation_failure() {
+    init_logger();
+    new_test_ext().execute_with(|| {
+        let code_id = CodeId::generate(&ProgramCodeKind::Default.to_bytes());
+        let pid = upload_program_default(USER_1, ProgramCodeKind::Default).unwrap();
+
+        run_to_block(2, None);
+
+        let mut old_version = 0;
+        let _reset_guard = DynamicSchedule::mutate(|schedule| {
+            // Insert new original code to cause re-instrumentation failure.
+            let wasm = ProgramCodeKind::Custom("(module)").to_bytes();
+            <<Test as Config>::CodeStorage as CodeStorage>::OriginalCodeStorage::insert(
+                code_id, wasm,
+            );
+
+            old_version = schedule.instruction_weights.version;
+            schedule.instruction_weights.version = 0xdeadbeef;
```

Comment:
instead of mutating schedule version you can just downgrade instrumentation version for code

Reply:
what the reason if I have already do like this?)

Reply:
Code simplicity for future generations :)
But you can keep it as it is ofc

Reply:
It's more simple solution, because downgrading version means that we are sure it's not 0. About 0xdeadbeaf we can be sure it won't be reached